src/qibo/models/circuit.py

```diff
@@ -512,7 +580,7 @@ def with_pauli_noise(self, noise_map: NoiseMapType):
                 noisy_circuit.add(noise_gate)
         return noisy_circuit
 
-    def add(self, gate):
+    def add(self, gate, position=-1):
```

Comment:
I think this feature is interesting. Is it better to introduce a new `insert` method like you did in `_Queue` on top of `list` or we want to change the `add` method like it is suggested here @qiboteam/coredevs?